test/wallet-http-test.js

```diff
@@ -269,20 +270,21 @@ describe('Wallet HTTP', function() {
 
     const opens = json.outputs.filter(output => output.covenant.type === types.OPEN);
     assert.equal(opens.length, 1);
+
+    // reset for next test
+    node.mempool.removeListener('tx', handler);
   });
 
   it('should create an open and not broadcast the transaction', async () => {
+    let entered = false;
+    const handler = () => entered = true;
+    node.mempool.on('tx', handler);
+
     const json = await wallet.createOpen({
       name: name,
       broadcast: false
     });
 
-    let entered = false;
-    node.mempool.on('tx', () => {
-      entered = true;
-      assert.ok(false);
-    });
-
     await sleep(500);
```

Comment:
It would be ideal to remove this sleep as well, but I'm not sure of a good way to do so